src/gpu/generic/sycl/group_normalization_kernel.hpp

```diff
@@ -0,0 +1,227 @@
+                            variance.get_pointer(),
+                            batch * conf_.num_groups + group_num);
+                }
+                ::sycl::group_barrier(it.get_group());
+                std_value = local_memory[it.get_local_range(0) + 1];
+            } else {
+                mean_value
+                        = load_float_value(data_type::f32, mean.get_pointer(),
+                                batch * conf_.num_groups + group_num);
+                std_value = load_float_value(data_type::f32,
+                        variance.get_pointer(),
+                        batch * conf_.num_groups + group_num);
+                std_value = ::sycl::sqrt(std_value + conf_.eta);
+            }
+
+            //Now start the normalization
+            for (dim_t j = it.get_local_linear_id(); j < num_elements_to_reduce;
+                    j += static_cast<dim_t>(it.get_local_range(0))) {
+                get_logical_index(src_wrapper.ndims(), dims, batch, group_num,
+                        num_spatial_elements, j, logical_index);
+                float value = load_float_value(src_wrapper.data_type(),
+                        src.get_pointer(), src_wrapper.off_v(logical_index));
+                float normalized_value = (value - mean_value) / std_value;
+                int32_t channel_value = (j / num_spatial_elements)
+                        % conf_.num_channels_per_group;
+                if (conf_.use_scale) {
+                    normalized_value *= load_float_value(data_type::f32,
+                            scale.get_pointer(),
+                            conf_.num_channels_per_group * group_num
+                                    + channel_value);
+                }
+                if (conf_.use_shift) {
+                    normalized_value += load_float_value(data_type::f32,
+                            shift.get_pointer(),
+                            conf_.num_channels_per_group * group_num
+                                    + channel_value);
+                }
+                if (conf_.src_scaling) {
+                    // Only one scaling factor per tensor is allowed,
+                    // as per the spec. Scaling factor will also always be f32 as per spec.
+                    normalized_value *= load_float_value(
+                            data_type::f32, src_scale.get_pointer(), 0);
+                }
+                float prev_value = normalized_value;
+                normalized_value = conf_.post_ops.apply(
+                        normalized_value, prev_value, po_args_, logical_index);
+                if (conf_.dst_scaling) {
+                    // Only one scaling factor per tensor is allowed,
+                    // as per the spec. Scaling factor will also always be f32 as per spec.
+                    normalized_value *= (1.0f
+                            / load_float_value(data_type::f32,
+                                    dst_scale.get_pointer(), 0));
+                }
```

Comment:
The [common check](https://github.com/oneapi-src/oneDNN/blob/main/src/common/group_normalization.cpp#L166) seems to only cover scales mask, not scales datatype.
Could you at least add a proper check in the `init()` for the scales datatype?